content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md

```diff
@@ -0,0 +1,207 @@
+                    ) %>%
+  # Generate the aggregate predictors used to fit the weights (in our example referred to as W')
+  
+  # average log income, retail price of cigarettes, and proportion of the
+  # population between 15 and 24 years of age from 1980 - 1988
+  generate_predictor(time_window = 1980:1988,
+                     ln_income = mean(lnincome, na.rm = T),
+                     ret_price = mean(retprice, na.rm = T),
+                     youth = mean(age15to24, na.rm = T)) %>%
+  # average beer consumption in the donor pool from 1984 - 1988
+  generate_predictor(time_window = 1984:1988,
+                     beer_sales = mean(beer, na.rm = T)) %>%
+  
+  # Lagged cigarette sales 
+  generate_predictor(time_window = 1975,
+                     cigsale_1975 = cigsale) %>%
+  generate_predictor(time_window = 1980,
+                     cigsale_1980 = cigsale) %>%
+  generate_predictor(time_window = 1988,
+                     cigsale_1988 = cigsale) %>%
+  
+   # Generate the fitted weights for the synthetic control
+  generate_weights(optimization_window = 1970:1988, # time to use in the optimization task
+                   margin_ipop = .02,sigf_ipop = 7,bound_ipop = 6 # optimizer options
+  ) %>%
+  
+  # Generate the synthetic control
+  generate_control()
+```
+{{% /codeblock %}}
+
+Once the synthetic control is generated, one can easily assess the fit by comparing the trends of the synthetic and observed time series. The idea is that the trends in the pre-intervention period should map closely onto one another. Let's verify that with a graph: 
+
+{{% codeblock %}}
+```R
+smoking_out %>% plot_trends()
+```
+{{% /codeblock %}}
+
+<p align = "center">
+<img src = "../images/1_synthetic.png" width="700">
+</p>
+
+As you can see, the pre-intervention trend between synthetic and observed cigarette sales is the same. 
+
+To capture the causal quantity (i.e. the difference between the observed and counterfactual), one can plot the differences using plot_differences(): 
+
+{{% codeblock %}}
+```R
+smoking_out %>% plot_differences()
+```
+{{% /codeblock %}}
+
+<p align = "center">
+<img src = "../images/2_synthetic.png" width="700">
+</p>
+
+Another useful way of evaluating the synthetic control is to look at how comparable the synthetic control is to the observed covariates of the treated unit by means of the following code: 
+
+{{% codeblock %}}
+```R
+smoking_out %>% grab_balance_table()
+```
+{{% /codeblock %}}
+ 
+## Advantages and limitations 
+
+According to Abadie (2021), the advantages of synthetic control methods are: 
+
+- SCMs provide a systematic way to estimate the effects of interventions when randomized controlled trials are not feasible. 
+- SCMs offer a transparent and data-driven method to construct the control group, which is based on observable characteristics and pre-treatment outcomes.
+- SCMs are flexible and can be applied in a wide range of contexts. 
+
+Possible disadvantages include: 
```

Comment:
Missing disadvantages:

- Bias in small samples: In cases of small pre-treatment periods (T0), large donor pools(J), and substantial noise, there is a higher risk of interpolation bias(i.e. the synthetic control is overly tailored to fit the idiosyncracies of the pre-treatment period) and overfitting.

- Inference: Traditional methods like t-tests and p-values don't apply well. Permutation tests are used which have limitations, especially with small donor pools and the assumption that any control unit could have been treated ( which will be violated if control units are very different from each other or the treated unit). This kind of inference is also computationally demanding.